app/src/main/java/ipvc/gymbuddy/app/adapters/TrainingPlanAdapter.kt

```diff
@@ -29,8 +32,19 @@ class TrainingPlanAdapter(dataset: List<TrainingPlan>): BaseRecyclerAdapter<Trai
     override fun bindViewHolder(holder: ViewHolder, item: TrainingPlan) {
         holder.name.text = item.name
         holder.editButton.setOnClickListener {
-            val bundle = bundleOf("trainingPlanId" to item.id)
-            holder.itemView.findNavController().navigate(R.id.trainer_trainingplans_update_fragment, bundle)
+            holder.itemView
+                .findNavController()
+                .navigate(
+                    R.id.trainer_trainingplans_update_fragment,
+                    bundleOf("trainingPlanId" to item.id)
+                )
+        }
+
+        holder.addClientButton.setOnClickListener {
+            val activity = it.context as FragmentActivity
+            val modalFragment = Modal.newInstance(R.layout.fragment_trainer_add_client_to_plan_modal)
+            modalFragment.setTitle(activity.getString(R.string.add_client_to, item.name))
+            modalFragment.show(activity.supportFragmentManager, "TrainerAddClientToPlanModalFragment")
```

Comment:
Recreate the modal fragment to add clients to the training plan by extending the Modal class.

then here it shouldn't be `Modal.newInstance`, it should be `AddClientToPlanModal(planId)`. 

Then all the logic regarding adding the client to the modal should be inside it. 

Use [material datepicker](https://github.com/material-components/material-components-android/blob/master/docs/components/DatePicker.md) to select the start and end date.